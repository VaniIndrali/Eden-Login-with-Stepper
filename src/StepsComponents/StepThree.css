.team-logo {
	height: 1.5rem;
	width: 1.5rem;
}

.mainText {
	margin-top: 3.125rem;
	padding: 0;
}

.subText {
	color: grey;
	margin-bottom: 1.875rem;
}

.btn-bd-primary {
	font-weight: 600;
	color: white;
	background-color: #744ab7;
	border-color: #744ab7;
	border-radius: 0.25rem;
	border: 0;
	padding: 0.625rem 7.938rem;
	align-items: center;
	width: 100%;
	text-align: center;
}

.Connector-d11-0-2-35:not(.active):not(.completed) {
	border-image: linear-gradient(
			45deg,
			#744ab7 0 50%,
			rgb(189, 189, 189) 50% 100%
		)
		1;
}

.Connector-d11-0-2-35.active {
	border-color: #744ab7 !important;
}

.Connector-d11-0-2-35.completed {
	border-color: #744ab7 !important;
}

.text-optional {
	color: grey;
}

#workspaceurl,
#workspaceurlexample {
	float: left;
	margin-bottom: 1rem;
}

#workspaceurlexample {
	width: 37%;
	border-radius: 0.25rem 0 0 0.25rem;
}

#workspaceurl {
	width: 63%;
	border-radius: 0 0.25rem 0.25rem 0;
}

#btn_myself,
#btn_team {
	text-align: start;
	padding: 1.25rem 0.625rem 1.25rem 1.25rem;
	border-radius: 0.5rem;
	border-color: rgb(189, 189, 189);
	border-width: 0.063rem;
	background-color: white;
	border-style: double;
	width: 50%;
}

#btn_myself {
	margin: 0.938rem 1.25rem 0.938rem 0;
}

#btn_myself:focus {
	outline-color: #744ab7;
}

.focused {
	fill: #744ab7;
}

#btn_team {
	margin: 0.938rem 0 0.938rem 1.25rem;
}

#btn_myself h6,
#btn_team h6 {
	color: grey;
}
