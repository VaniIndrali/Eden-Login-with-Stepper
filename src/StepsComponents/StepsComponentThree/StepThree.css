.team-logo {
	height: 1.5rem;
	width: 1.5rem;
}

.Connector-d11-0-2-35:not(.active):not(.completed).disabled {
	border-image: linear-gradient(
			45deg,
			#744ab7 0 50%,
			rgb(189, 189, 189) 50% 100%
		)
		1;
}

.Connector-d11-0-2-35.active {
	border-color: #744ab7 !important;
}

.Connector-d11-0-2-35.completed {
	border-color: #744ab7 !important;
}

.text-optional {
	color: grey;
}

#btn_myself,
#btn_team {
	text-align: start;
	padding: 1.25rem 0.625rem 1.25rem 1.25rem;
	border-radius: 0.5rem;
	border-color: rgb(189, 189, 189);
	border-width: 0.063rem;
	background-color: white;
	border-style: double;
	width: 50%;
}

#btn_myself {
	margin: 0.938rem 1.25rem 0.938rem 0;
}

#btn_myself,
#btn_team:focus {
	outline-color: #744ab7;
}

.focused {
	fill: #744ab7;
}

#btn_team {
	margin: 0.938rem 0 0.938rem 1.25rem;
}

#btn_myself h6,
#btn_team h6 {
	color: grey;
}
