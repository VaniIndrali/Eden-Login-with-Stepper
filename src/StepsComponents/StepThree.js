import { React } from "react";
import "./StepThree.css";

export const StepThree = (props) => {
	const handleSubmit = (e) => {
		e.preventDefault();
		props.onSubmit();
	};

	const addFocusCss = () => {
		var icon = document.getElementById("svgPerson");
		icon.classList.add("focused");
	};

	const addBlurCss = () => {
		var icon = document.getElementById("svgPerson");
		icon.classList.remove("focused");
	};

	return (
		<div id="loginThree">
			<div className="container">
				<h6 className="text-center subText">
					We'll streamline your setup experience accordingly.
				</h6>
				<div
					id="login-row"
					className="row justify-content-center align-items-center"
				>
					<div
						id="login-column"
						className="col-md-11"
					>
						<div
							id="login-box"
							className="col-md-12"
						>
							<form
								id="login-form"
								className="form"
								onSubmit={(e) => handleSubmit(e)}
							>
								<div className="form-group">
									<div class="btn-group">
										<button
											id="btn_myself"
											type="button"
											onFocus={() => addFocusCss()}
											onBlur={() => addBlurCss()}
										>
											<svg
												xmlns="http://www.w3.org/2000/svg"
												width="27"
												height="27"
												fill="currentColor"
												class="bi bi-person-fill"
												viewBox="0 0 16 14"
												id="svgPerson"
											>
												<path d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z" />
											</svg>
											<h5 className="mt-2">For myself</h5>
											<h6>
												Write better. Think more
												clearly. Stay organized.
											</h6>
										</button>
										<button
											id="btn_team"
											type="button"
										>
											<img
												src="https://i.imgur.com/op4AsSO.png"
												className="team-logo"
												alt="team"
											/>
											<h5 className="mt-3">
												With my team
											</h5>
											<h6>
												Wikis, docs, tasks & projects,
												all in one place.
											</h6>
										</button>
									</div>
								</div>
								<div className="form-group">
									<input
										type="submit"
										name="Create Workspace"
										className="btn-bd-primary"
										value="Create Workspace"
									/>
								</div>
							</form>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};

export default StepThree;
