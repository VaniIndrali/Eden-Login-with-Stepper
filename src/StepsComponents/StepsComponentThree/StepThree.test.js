import { render, screen } from "@testing-library/react";
import StepThree from "./StepThree";

test("renders StepThree", () => {
	const props = {
		onSubmit: jest.fn(),
	};

	const component = render(<StepThree {...props} />);

	expect(
		screen.getByRole("heading", {
			name: "We'll streamline your setup experience accordingly.",
		})
	).toBeTruthy();

	expect(
		screen.getByRole("button", {
			name: "For myself Write better. Think more clearly. Stay organized.",
		})
	).toBeTruthy();
	expect(
		screen.getByRole("button", {
			name: "team With my team Wikis, docs, tasks & projects, all in one place.",
		})
	).toBeTruthy();
	expect(
		screen.getByRole("button", {
			name: "Create Workspace",
		})
	).toBeTruthy();

	expect(component).toMatchSnapshot();
});
