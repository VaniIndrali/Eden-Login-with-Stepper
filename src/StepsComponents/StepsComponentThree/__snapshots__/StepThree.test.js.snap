// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StepThree 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="loginThree"
      >
        <div
          class="container col-md-12 col-9"
        >
          <h6
            class="text-center subText"
          >
            We'll streamline your setup experience accordingly.
          </h6>
          <div
            class="row justify-content-center align-items-center"
            id="login-row"
          >
            <div
              class="col-md-11"
              id="login-column"
            >
              <div
                class="col-md-12"
                id="login-box"
              >
                <form
                  class="form"
                  id="login-form"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="btn-group"
                    >
                      <button
                        id="btn_myself"
                        type="button"
                      >
                        <svg
                          class="bi bi-person-fill"
                          fill="currentColor"
                          height="27"
                          id="svgPerson"
                          viewBox="0 0 16 14"
                          width="27"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
                          />
                        </svg>
                        <h5
                          class="mt-2"
                        >
                          For myself
                        </h5>
                        <h6>
                          Write better. Think more clearly. Stay organized.
                        </h6>
                      </button>
                      <button
                        id="btn_team"
                        type="button"
                      >
                        <img
                          alt="team"
                          class="team-logo"
                          src="https://i.imgur.com/op4AsSO.png"
                        />
                        <h5
                          class="mt-3"
                        >
                          With my team
                        </h5>
                        <h6>
                          Wikis, docs, tasks & projects, all in one place.
                        </h6>
                      </button>
                    </div>
                  </div>
                  <div
                    class="form-group"
                  >
                    <input
                      class="button"
                      name="Create Workspace"
                      type="submit"
                      value="Create Workspace"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="loginThree"
    >
      <div
        class="container col-md-12 col-9"
      >
        <h6
          class="text-center subText"
        >
          We'll streamline your setup experience accordingly.
        </h6>
        <div
          class="row justify-content-center align-items-center"
          id="login-row"
        >
          <div
            class="col-md-11"
            id="login-column"
          >
            <div
              class="col-md-12"
              id="login-box"
            >
              <form
                class="form"
                id="login-form"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="btn-group"
                  >
                    <button
                      id="btn_myself"
                      type="button"
                    >
                      <svg
                        class="bi bi-person-fill"
                        fill="currentColor"
                        height="27"
                        id="svgPerson"
                        viewBox="0 0 16 14"
                        width="27"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
                        />
                      </svg>
                      <h5
                        class="mt-2"
                      >
                        For myself
                      </h5>
                      <h6>
                        Write better. Think more clearly. Stay organized.
                      </h6>
                    </button>
                    <button
                      id="btn_team"
                      type="button"
                    >
                      <img
                        alt="team"
                        class="team-logo"
                        src="https://i.imgur.com/op4AsSO.png"
                      />
                      <h5
                        class="mt-3"
                      >
                        With my team
                      </h5>
                      <h6>
                        Wikis, docs, tasks & projects, all in one place.
                      </h6>
                    </button>
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <input
                    class="button"
                    name="Create Workspace"
                    type="submit"
                    value="Create Workspace"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
